test/decompiler/reference/engine/game/video_REF.gc

```diff
@@ -0,0 +1,111 @@
+     (set! (-> *video-parms* relative-y-scale) 1.0)
+     (set! *ticks-per-frame* 9765)
+     (set! (-> *math-camera* isometric vector 1 y) 0.5)
+     (set! (-> *math-camera* y-pix) 112.0)
+     (set! (-> *math-camera* y-clip) 448.0)
+     (set! (-> *shadow-data* texoffset y) 112.5)
+     )
+    ((= v1-0 'pal)
+     (set! (-> *video-parms* screen-sy) 256)
+     (set! (-> *setting-control* default screenx) 0)
+     (set! (-> *setting-control* default screeny) 24)
+     (set! (-> *video-parms* screen-pages-high) 8)
+     (set! (-> *video-parms* relative-y-scale) 1.1428572)
+     (set! *ticks-per-frame* #x2dc6)
+     (set! (-> *math-camera* isometric vector 1 y) 0.4375)
+     (set! (-> *math-camera* y-pix) 128.0)
+     (set! (-> *math-camera* y-clip) 512.0)
+     (set! (-> *shadow-data* texoffset y) 128.5)
+     )
+    )
+   )
+  (set-time-ratios *display* (-> *display* time-ratio))
+  (set! (-> *video-parms* reset-video-mode) #t)
+  (set! (-> *video-parms* screen-hy) (/ (-> *video-parms* screen-sy) 2))
+  (set! (-> *video-parms* screen-miny) (- 2048 (-> *video-parms* screen-hy)))
+  (set! (-> *video-parms* screen-maxy) (+ (-> *video-parms* screen-hy) 2048))
+  (set! (-> *video-parms* screen-masky) (+ (-> *video-parms* screen-sy) -1))
+  (set!
+   (-> *pause-context* origin y)
+   (the float (+ (-> *video-parms* screen-sy) -54))
+   )
+  (set! (-> *pause-context* height) (the float (-> *video-parms* screen-sy)))
+  (set!
+   (-> *font-default-matrix* vector 1 y)
+   (-> *video-parms* relative-y-scale)
+   )
+  (set!
+   (-> *font-default-matrix* vector 3 y)
+   (- (the float (-> *video-parms* screen-hy)))
+   )
+  (set! (-> *video-parms* relative-y-scale-reciprical) 1.0)
+  (set! *profile-y* (+ (-> *video-parms* screen-miny) 8))
+  (set! (-> *video-parms* set-video-mode) #t)
+  (set-hud-aspect-ratio (get-aspect-ratio) arg0)
+  (if *progress-process*
+   (TODO-RENAME-23 (-> *progress-process* 0) (get-aspect-ratio))
+   )
+  (let ((v0-5 0))
+   )
+  (none)
+  )
+
+;; definition for function get-video-mode
+(defun get-video-mode ()
+  (-> *setting-control* current video-mode)
+  )
+
+;; definition for function set-aspect-ratio
+;; INFO: Return type mismatch int vs none.
+(defun set-aspect-ratio ((arg0 symbol))
+  (let ((v1-0 arg0))
+   (cond
+    ((= v1-0 'aspect4x3)
+     (set! (-> *video-parms* relative-x-scale) 1.0)
+     (set! (-> *video-parms* relative-x-scale-reciprical) 1.0)
+     )
+    ((= v1-0 'aspect16x9)
+     (set! (-> *video-parms* relative-x-scale) 0.75)
+     (set! (-> *video-parms* relative-x-scale-reciprical) 1.3333334)
+     )
+    )
+   )
+  (set!
+   (-> *font-default-matrix* vector 0 x)
+   (-> *video-parms* relative-x-scale)
+   )
+  (set-hud-aspect-ratio arg0 (the-as video-parms (get-video-mode)))
+  (when *progress-process*
+   (let* ((s5-1 (-> *progress-process* 0))
+          (s4-1 (method-of-object s5-1 TODO-RENAME-23))
+          )
+    (get-video-mode)
+    (s4-1 s5-1 arg0)
```

Comment:
Hmm, this still doesn't look quite right.  I'd expect this to look something like:
```lisp
(TODO-RENAME-23 (-> *progress-process* 0) s5-1 arg0 (get-video-mode))
```
I think this function takes an additional argument.

This is a case where it might help to look at the assembly for the method call. The GOAL compiler generally evaluates expressions left-to-right:

First, it gets the object that it will call the method on: `(-> *progress-process*)`
```
    lw v1, *progress-process*(s7)
    lwu s5, 0(v1)
```

Then, it gets the method from the method table:
```
    lwu v1, -4(s5)            ;; (-> s5 type)
    lwu s4, 108(v1)           ;; (-> v1 method-table 23)
```

The first two arguments are just variables and don't require any code to evaluate.  So this is code for the third argument: `(get-video-mode)`
```
    lw t9, get-video-mode(s7) ;; (set! t9 get-video-mode)[39]
    jalr ra, t9               ;; (call!)[40]
    sll v0, ra, 0
```
Next, it moves the result into the argument register:
```
    or a2, v0, r0             ;; (set! a2 v0)[41]
```

And the final step is to insert a bunch of moves that move existing variables into the right registers (the compiler may omit these if the variable happens to be in the right register already):
```
    or t9, s4, r0             ;; (set! t9 s4)[42]
    or a0, s5, r0             ;; (set! a0 s5)[43]
    or a1, gp, r0             ;; (set! a1 gp)[44]
```
and then call the method
```
    jalr ra, t9               ;; (call!)[45]
    sll v0, ra, 0
```